hashintel/hash
# Disable Edit property/link type buttons for types that belong to other users/orgs

## 🌟 What is the purpose of this PR?



This PR disables the "edit property type" and "edit link type" buttons when the property type/link type doesn't belong to the user.

## 🔗 Related links





- [Asana task](https://app.asana.com/0/1203358502199087/1204081024577367/f) _(internal)_

## 🔍 What does this change?




- Added `canEditResource` to the ontologyFunctions passed to the entity type editor;
- Called it to check if the "Edit property type"/"Edit link type" buttons should be disabled;
- Added a tooltip to explain that the button is the disabled because the type doesn't belong to the user.

## Pre-Merge Checklist 🚀

### 🚢 Has this modified a publishable library?




This PR:


### 📜 Does this require a change to the docs?




The changes in this PR:


### 🕸️ Does this require a change to the Turbo Graph?




The changes in this PR:



## 🐾 Next steps


- Apply changes to the entity type editor in Blockprotocol.


## ❓ How to test this?



1.  Create an entity type;
2. Create a new property type and check if you can edit it;
3. Add an existing property type from another namespace and check if the edit button is disabled;
4. Do the same for link types.

## 📹 Demo


![image](https://user-images.githubusercontent.com/37453800/236024269-b8b4b1e6-8d4e-4062-bf62-e7ab9fa6381c.png)

![image](https://user-images.githubusercontent.com/37453800/236024359-ee72ebf2-4ab9-4b8d-9fab-43648970786f.png)
